modules/shared-filesystem-utils/src/main/java/org/opencastproject/assetmanager/util/AssetPathUtils.java

```diff
@@ -130,7 +130,7 @@ public static File getLocalFile(final List<String> localPaths, final String orga
     final String version = assetPath[5];
     final String filename = mediaPackageElementID + '.' + FilenameUtils.getExtension(assetPath[6]);
     // Check under which path the mediapackage exists
-    String localPath = null;
+    String localPath = localPaths.get(0);
```

Comment:
This seems wrong and feels like this will surely have side effects. The function accepts a list of potential local paths, iterating over the list and filters them using some criteria. With your patch, the first entry just skips all checks and requirements, so that the results does no longer guarantee these properties.

I don't know this code well, but I would expect `null` to be a valid return value if there is no valid local path for the (remote) file.

Reply:
`null` is not valid. Line 133 initializes the value to `null`. Lines 134-139, check each path in `localPaths` to see if the mediapackage folder exists in that path. The Line 140 assumes that the loop actually found a path, which is not a valid assumption in the case of remote asset managers. This line will throw an exception if `localPath` is `null`. Line 141 then again checks if the actual file exists. For this reason, it is ok to set the default `localPath` to the first one.

Reply:
We discussed this in the dev meeting and agreed that this approach is confusing. I changed the path to a simple null check.